frontend/packages/console-shared/src/components/actions/menu/ActionMenuItem.tsx

```diff
@@ -1,15 +1,117 @@
+  autoFocus?: boolean;
+  /** A short description of the dropdown item, displayed under the dropdown item content */
+  description?: React.ReactNode;
+  /** Events to prevent when the item is disabled */
+  inoperableEvents?: string[];
+}
+
+// Duplicated from https://github.com/patternfly/patternfly-react/blob/5489a9031c8a58f7ac4f91f6d0fe1c86b0dbb0bf/packages/react-core/src/deprecated/components/Dropdown/DropdownItem.tsx#L6-L52
+interface DropdownItemProps extends Omit<InternalDropdownItemProps, 'tabIndex'>, OUIAProps {
+  /** Anything which can be rendered as dropdown item */
+  children?: React.ReactNode;
+  /** Classes applied to root element of dropdown item */
+  className?: string;
+  /** Class to be applied to list item */
+  listItemClassName?: string;
+  /**
+   * A ReactElement to render, or a string to use as the component tag.
+   * Example: component={<Link to="/components/alert/">Alert</Link>}
+   * Example: component="button"
+   * If React.isValidElement(component) the className prop will be injected unless styleChildren="false"
+   */
+  component?: React.ReactNode;
+  /** ID for the component element */
+  componentID?: string;
+  /** Whether to set className on component when React.isValidElement(component) */
+  styleChildren?: boolean;
+  /** Render dropdown item as disabled option */
+  isDisabled?: boolean;
+  /** Render dropdown item as aria-disabled option */
+  isAriaDisabled?: boolean;
+  /** Render dropdown item as non-interactive item */
+  isPlainText?: boolean;
+  /** Default hyperlink location */
+  href?: string;
+  /** Tooltip to display when hovered over the item */
+  tooltip?: React.ReactNode;
+  /** Additional tooltip props forwarded to the Tooltip component */
+  tooltipProps?: any;
+  /** Additional node to include alongside item within the <li> */
+  additionalChild?: React.ReactNode;
+  /** Custom item rendering that receives the DropdownContext */
+  customChild?: React.ReactNode;
+  /** tabIndex to use, null to unset it */
+  tabIndex?: number | null;
+  /** An image to display within the DropdownItem, appearing before any component children */
+  icon?: React.ReactNode;
+  /** Initial focus on the item when the menu is opened (Note: Only applicable to one of the items) */
+  autoFocus?: boolean;
+  /** A short description of the dropdown item, displayed under the dropdown item content */
+  description?: React.ReactNode;
+  /** Value to overwrite the randomly generated data-ouia-component-id. */
+  ouiaId?: number | string;
+  /** Set the value of data-ouia-safe. Only set to true when the component is in a static state, i.e. no animations are occurring. At all other times, this value must be false. */
+  ouiaSafe?: boolean;
+}
+
 export type ActionMenuItemProps = {
   action: Action;
-  component?: React.ComponentType<DropdownItemPropsDeprecated>;
+  component?: React.ComponentType<DropdownItemProps>;
```

Comment:
Can't we just use the [non-deprecated `DropdownItemProps`](https://github.com/patternfly/patternfly-react/blob/c1ab963828ff4dbd92ac3047453ef242a56c16ee/packages/react-core/src/components/Dropdown/DropdownItem.tsx#L10) instead duplicating the old one?

From the usages of the `component` prop I can find, one of them already takes the non-deprecated one: 

https://github.com/openshift/console/blob/859242b8c0993e20ef73698e899babd316062832/frontend/packages/topology/src/actions/contextMenuActions.tsx#L40

Reply:
I was assuming not since when the migration to PatternFly 5 was done the decision was made to use the deprecated ones.  But you may be right...

Reply:
It might've just been a blanket find-and-replace to avoid risk on a big PR

Reply:
That is probably a good assumption.

Reply:
Updated.  Thanks, @logonoff!